### PR TITLE
refactor: replace react-scripts (cra) with vite, and make necessary changes

### PR DESCRIPTION
**Pull request by sebastianvitterso**
## What does this pull request change, and why?
Create React App (CRA) isn't as modern anymore, and using vite is (in my opinion) a cleaner choice.

Also we've made the switch in most of our other projects, so it's about time we do it here too. 

Doesn't really matter though, since we're switching only for the example project, just thought it would be nice.

## Issues related to this change
None

### CODE REVIEW

File: index.html
Comment (sebastianvitterso) [inline index.html:1]: Not sure why this file didn't register as being renamed/moved. It's the same `index.html`, just with the script tag engrained.

File: package.json [hunk 1]
```diff
@@ -2,27 +2,28 @@
...
-    "build": "react-scripts build",
+    "start": "vite",
+    "build": "vite build",
     "test": "jest --silent",
     "test:watch": "jest --silent --watch"
   },
   "dependencies": {
     "react": "18.3.1",
-    "react-dom": "18.3.1",
-    "react-scripts": ">=5.0.1"
```
Comment (sebastianvitterso) [inline package.json:13]: Good riddance.

File: package.json [hunk 2]
```diff
@@ -2,27 +2,28 @@
...
+    "@types/jest": "^29.5.14",
+    "@types/react": "18.3.12",
+    "@vitejs/plugin-react": "^4.3.3",
     "jest-environment-jsdom": "^29.7.0",
-    "ts-jest": "^29.1.2",
-    "typescript": "5.4.5"
+    "jest": "^29.7.0",
+    "ts-jest": "^29.2.5",
+    "typescript": "5.6.3",
+    "vite": "^5.4.10"
```
Comment (sebastianvitterso) [inline package.json:26]: Took the opportunity to update packages.

File: src/index.jsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ const authConfig = {
   authorizationEndpoint: 'https://keycloak.ofstad.xyz/realms/master/protocol/openid-connect/auth',
   tokenEndpoint: 'https://keycloak.ofstad.xyz/realms/master/protocol/openid-connect/token',
   logoutEndpoint: 'https://keycloak.ofstad.xyz/realms/master/protocol/openid-connect/logout',
-  redirectUri: 'http://localhost:3000/',
+  redirectUri: 'http://localhost:5173/',
```
Comment (sebastianvitterso) [inline src/index.jsx:20]: Vite uses the port `VITE` by default.

File: src/index.jsx [hunk 4]
```diff
@@ -45,10 +45,10 @@ function LoginInfo() {
       </>
       {token ? (
         <>
-          <button onClick={() => logOut('rememberThis', idTokenData.session_state)}>Log out</button>
+          <button onClick={() => logOut('rememberThis', idTokenData?.session_state)}>Log out</button>
```
Comment (sebastianvitterso) [inline src/index.jsx:50]: `idTokenData` is potentially `null`.

File: src/index.jsx [hunk 5]
```diff
@@ -45,10 +45,10 @@ function LoginInfo() {
...
           <span style={{ margin: '0 10px' }}>
             Access token will expire at:{' '}
-            {new Date(localStorage.getItem('ROCP_tokenExpire') * 1000).toLocaleTimeString()}
+            {new Date(Number(localStorage.getItem('ROCP_tokenExpire')) * 1000).toLocaleTimeString()}
```
Comment (sebastianvitterso) [inline src/index.jsx:53]: `localStorage` actually only saves strings, so if we store a number, we should explicitly convert it to a number before doing arithmetics on it.

File: src/index.jsx [hunk 6]
```diff
@@ -110,6 +110,7 @@ function LoginInfo() {
 }
 
 const container = document.getElementById('root')
+if (!container) throw new Error('No container found')
```
Comment (sebastianvitterso) [inline src/index.jsx:115]: `document.getElementById` may return `null`. This assumes it doesn't.

File: tests/auth-util.test.ts [hunk 7]
```diff
@@ -10,7 +10,7 @@ const authConfig: TInternalConfig = {
   clientId: 'myClientID',
   authorizationEndpoint: 'myAuthEndpoint',
   tokenEndpoint: 'myTokenEndpoint',
-  redirectUri: 'http://localhost:3000/',
+  redirectUri: 'http://localhost:5173/',
```
Comment (sebastianvitterso) [inline tests/auth-util.test.ts:13]: Not sure if I needed to change this one.

File: index.html
Comment (sebastianvitterso) [inline index.html:1]: Not sure why this is being marked as a new file, it's just moved (and added one line for the script).

**Comment by soofstad** [comment]
Seems to work! :tada:

**Pull request merged**